x-pack/plugins/ml/public/application/jobs/new_job/pages/job_type/page.tsx

```diff
@@ -68,25 +74,27 @@ export const Page: FC = () => {
     },
   };
 
-  const getUrl = (basePath: string) => {
+  const getUrlParams = () => {
     return !isSavedSearchSavedObject(currentSavedSearch)
-      ? `${basePath}?index=${currentIndexPattern.id}`
-      : `${basePath}?savedSearchId=${currentSavedSearch.id}`;
+      ? `?index=${currentIndexPattern.id}`
+      : `?savedSearchId=${currentSavedSearch.id}`;
   };
 
+  const navigateToPath = (path: string | undefined) =>
+    navigateToUrl(getUrlForApp('ml', { path: `${path}${getUrlParams()}` }));
```

Comment:
could this be wrapped in a provider function that just takes the path and supplied by `useMlContext `? 
we it should also use `PLUGIN_ID`  from `kibana/x-pack/plugins/ml/common/constants/app.ts`

Reply:
Yes, it's something I'd like to do in a follow up when I'll migrate all internal urls away from using hashes. The original scope of this PR was to introduce the updated router and let the existing urls be redirected, so this is more of a short term workaround for broken links for this PR.

Reply:
As discussed, decided to do it in this PR, we now have `useNavigateToPath()` based on `useMlContext()`, available via `contexts/kibana`. Update in fad85ec.